hashes/src/macros.rs

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: CC0-1.0
+
+//! Public macros.
+
+/// Adds trait impls to a bytelike type.
+///
+/// Implements:
+///
+/// * `str::FromStr`
+/// * `fmt::{LowerHex, UpperHex}` using `hex-conservative`.
+/// * `fmt::{Display, Debug}` by calling `LowerHex`
+/// * `serde::{Deserialize, Serialize}`
+/// * `AsRef[u8; $len]`
+/// * `AsRef[u8]`
+/// * `Borrow<[u8]>`
+/// * `slice::SliceIndex<[u8]>`
```

Comment:
I believe we've agreed elsewhere that we don't want this impl.

Reply:
I don't remember but I've removed it (and menally noted it). Thanks

Reply:
Removing it breaks the `serde` logic. Just leaving it in for now.

Reply:
Can't `serde` simply call `as_ref` instead of indexing? Should be a trivial change.